internal/pkg/service/buffer/worker/distribution/work.go

```diff
@@ -19,7 +21,10 @@ import (
 // The DistributedWork callback is called:
 //   - On initialization, after the Node.StartWork method is called.
 //   - On distribution change.
-//   - Periodically, according to the executorConfig.restartInterval.
+//   - Periodically, according to the workConfig.restartInterval.
+//
+// Before each invocation of the callback, for the reasons listed above, the previous context is canceled.
+// So invocation of the callback can be understood as a reset event, because the previous work is terminated.
 type DistributedWork func(ctx context.Context, assigner *Assigner) (initDone <-chan error)
```

Comment:
Tu som doplnil komentar, ale ako som pisal, tak uvazujem, ze ci tuto cast este potrebujeme, aby sa na tom uz nikto "nenachytal".

Reply:
Riesi to situaciu pri `tasks/orchestrator`, ked nejaka worker node vypadne, tak sa to musi zrestartovat, aby jej tasks prevzali nejake ine nodes. Pri cleanup to nie je problem, kedze ten nie je kriticky a spusta sa pravidelne.

Reply:
Mozno to nejak premenovat `distribution.StartWork` -> `distribution.RestartOnChange`? Neviem ...